multichain-testing/test/auto-stake-it.test.ts

```diff
@@ -0,0 +1,293 @@
+  const { portId, channelId } =
+    chainInfo[destination.chainName].connections[senderChainInfo.chain.chain_id]
+      .transferChannel;
+
+  /** @returns {bigint} nanosecond timestamp 5 mins in the future */
+  const getTimeoutTimestampNs = () =>
+    (BigInt(new Date().getTime()) + SECONDS_PER_MINUTE * 5n) *
+    NANOSECONDS_PER_MILLISECOND;
+
+  const msgTransfer = MsgTransfer.fromPartial({
+    sender: sender.address,
+    receiver: destination.address,
+    token: { denom: amount.denom, amount: String(amount.value) },
+    sourcePort: portId,
+    sourceChannel: channelId,
+    timeoutHeight,
+    timeoutTimestamp: timeoutHeight
+      ? undefined
+      : timeoutTimestamp ?? getTimeoutTimestampNs(),
+    memo,
+  });
+  const { fee_tokens } = senderChainInfo.chain.fees ?? {};
+  if (!fee_tokens || !fee_tokens.length) {
+    throw Error('no fee tokens in chain config for' + sender.chainName);
+  }
+  const { high_gas_price, denom } = fee_tokens[0];
+  if (!high_gas_price) throw Error('no high gas price in chain config');
+  const fee = makeFeeObject({
+    denom: denom,
+    gas: 150000,
+    gasPrice: high_gas_price,
+  });
+
+  return [
+    msgTransfer.sender,
+    msgTransfer.receiver,
+    msgTransfer.token,
+    msgTransfer.sourcePort,
+    msgTransfer.sourceChannel,
+    msgTransfer.timeoutHeight,
+    Number(msgTransfer.timeoutTimestamp),
+    fee,
+    msgTransfer.memo,
+  ];
+};
+
+const createFundedWalletAndClient = async (
+  t: ExecutionContext,
+  chainName: ChainName,
+) => {
+  const { chain, creditFromFaucet, getRpcEndpoint } = useChain(chainName);
+  const wallet = await createWallet(chain.bech32_prefix);
+  const address = (await wallet.getAccounts())[0].address;
+  t.log(`Requesting faucet funds for ${address}`);
+  await creditFromFaucet(address);
+  const client = await SigningStargateClient.connectWithSigner(
+    getRpcEndpoint(),
+    wallet,
+  );
+  return { client, wallet, address };
+};
+
+test('auto-stake-it: creates accts, register tap, transfer to auto-delegate', async t => {
+  const { wallets, makeQueryTool, provisionSmartWallet, retryUntilCondition } =
+    t.context;
+
+  const fundAndTransfer = async (
+    chainName: ChainName,
+    agoricAddr: string,
+    amount = 100n,
+  ) => {
+    const { staking } = useChain(chainName).chainInfo.chain;
+    const denom = staking?.staking_tokens[0].denom;
+    if (!denom) throw Error(`no denom for ${chainName}`);
+
+    const { client, address, wallet } = await createFundedWalletAndClient(
+      t,
+      chainName,
+    );
+    await sleep(1500); // wait for wallet to instantiate
+    console.log('Balances:', await client.getAllBalances(address));
+
+    const transferArgs = makeIBCTransferMsg(
+      { denom, value: amount },
+      { address: agoricAddr, chainName: 'agoric' },
+      { address: address, chainName },
+    );
+    // @ts-expect-error spread argument for concise code
+    const txRes = await client.sendIbcTokens(...transferArgs);
```

Comment:
`sendIbcTokens` is deprecated, since it can't supply a memo field on the MsgTransfer (just the outer tx). Only using it here since `client.simulate()` and `client.signAndBroadcast()` kept yielding:
```
invalid source port ID: identifier cannot be blank: invalid ident
```
It would seem like an obvious error to address, but appears to be more of a red herring as sourcePort was definitely provided. I suspected a mismatch between the client registry and the encoder (`@agoric/cosmic-proto`), but registering a custom messge did not seem to do the trick either. I similarly had no luck with `cosmjs-types`, so maybe something else.

An approach maybe worth exploring is using the RPC client from telescope via `@agoric/cosmic-proto`.